### PR TITLE
chore: add rsdoctor's examples

### PR DESCRIPTION
**Pull request by easy1090**
chore: add rsdoctor's examples:
- rsbuild
- modern.js
- next.js

### CODE REVIEW

File: rsdoctor/modern-minimal/modern.config.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { appTools, defineConfig } from '@modern-js/app-tools';
+import { RsdoctorWebpackPlugin } from '@rsdoctor/webpack-plugin';
```
Comment (chenjiahan) [inline rsdoctor/modern-minimal/modern.config.ts:2]: we can use the Rspack mode of Modern.js in this example

Comment (chenjiahan) [reply rsdoctor/modern-minimal/modern.config.ts:2]: (which is faster

File: rsdoctor/next-minimal/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "next-app",
```
Comment (chenjiahan) [inline rsdoctor/next-minimal/package.json:2]: there are many examples in this repo, so we'd better set a unified name for all Rsdoctor examples, such as `rsdoctor-next-app`.

File: rsdoctor/rsbuild-minimal/rsbuild.config.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      chain.plugin('Rsdoctor').use(RsdoctorRspackPlugin, [
+        {
+          disableClientServer: !process.env.ENABLE_CLIENT_SERVER,
+          features: ['bundle', 'plugins', 'loader', 'resolver']
+        },
+      ]);
+    }
+  },
+  output: {
+    disableFilenameHash: true,
```
Comment (chenjiahan) [inline rsdoctor/rsbuild-minimal/rsbuild.config.ts:18]: why do we need to disable hash and minimize

Comment (easy1090) [reply rsdoctor/rsbuild-minimal/rsbuild.config.ts]: done~

File: rsdoctor/next-minimal/package.json (hunk 2 above)
Comment (easy1090) [reply rsdoctor/next-minimal/package.json]: done~

File: rsdoctor/modern-minimal/modern.config.ts (hunk 1 above)
Comment (easy1090) [reply rsdoctor/modern-minimal/modern.config.ts]: change to rspack mode, rspack mode faster more.

**Review by chenjiahan — APPROVED** [review]
👍

**Pull request merged**